subxt/src/utils/unchecked_extrinsic.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! The "default" Substrate/Polkadot UncheckedExtrinsic.
+//! This is used in codegen for runtime API calls.
+//!
+//! The inner bytes represent the encoded extrinsic expected by the
+//! runtime APIs. Deriving `EncodeAsType` would lead to the inner
+//! bytes to be re-encoded (length prefixed).
+
+use std::marker::PhantomData;
+
+use codec::Decode;
+use scale_decode::{visitor::DecodeAsTypeResult, IntoVisitor, Visitor};
+
+/// The unchecked extrinsic from substrate.
+#[derive(Clone, Debug, Eq, PartialEq, Decode)]
+pub struct UncheckedExtrinsic<Address, Call, Signature, Extra>(
+    pub Vec<u8>,
+    #[codec(skip)] pub PhantomData<(Address, Call, Signature, Extra)>,
+);
```

Comment:
Potentially we could reuse some other code more by having this be:

```rust
pub struct UncheckedExtrinsic<Address, Call, Signature, Extra>{
    encoded: Static<Encoded>,
    #[codec(skip)] pub PhantomData<(Address, Call, Signature, Extra)>,
};
```

Then, all of the impls could delegate to the `Static<Encoded>` type, which I think already does what we want it to.

In any case though, let's give this a nicer interface, something like:

```rust
// make a new one:
UncheckedExtrinsic::new(bytes)
// can convert bytes into this type easily:
impl From<Vec<u8>> for UncheckedExtrinsic
```

Hopefully type inference will sort out the generic params when it's used, ie

```rust
let runtime_api_call = node_runtime::apis()
        .transaction_payment_api()
        .query_fee_details(tx_bytes.into(), len);
```

Will hopefully Just Work, and infer all of the types properly!